src/web/core/src/main/java/org/geoserver/web/data/store/DefaultDataStoreEditPanel.java

```diff
@@ -160,13 +160,15 @@ private Panel getInputComponent(final String componentId, final IModel paramsMod
         final Class<?> binding = paramMetadata.getBinding();
         final List<Serializable> options = paramMetadata.getOptions();
 
+        boolean visible = true;
+
         Panel parameterPanel;
         if("dbtype".equals(paramName) || "filetype".equals(paramName)) {
```

Comment:
That also requires changing the GeoTools datastore factory parameters, as they are not using said flag:
public static final Param DBTYPE = new Param("dbtype", String.class, "Type", true);
public static final Param FILE_TYPE = new Param("filetype", String.class,
            "Discriminator for directory stores", false, "shapefile", new KVP(Param.LEVEL,
                    "advanced"));